### ISSUE TITLE
Dynamically creating a Text Image and overlaying it on a Gif

### ISSUE THREAD

**Issue by Zueripat** [lovell/sharp#3517]
### What are you trying to achieve?
I am trying to use the the new `Create Image from Text` feature often talked about in Issue #512 and Mentioned in the [Documentation](https://sharp.pixelplumbing.com/api-constructor) to Create a overlay Image using Text provided to my Script and compose this Image onto a Animated Image. 

#### Questions
How can i rezize the Text overlay Image without changing the size of the Text size?
Why is it, that when i just change the width and height of the Text to match the Image the Text does not display on a fixed position, but rather in a scrolling animation?

### When you searched for similar issues, what did you find that might be related?
How to apply an overlay on each frame of a Gif: #3347 
Compositing PNG o GIF: #3435
How to overlayWith new sharp() then how to overlayWith text: #1263
...

### Please provide a minimal, standalone code sample, without other dependencies, that demonstrates this question
**overlayTextOnImage Function**
```typescript
const overlayTextOnImage = async (text: string, image: Sharp) => {
    const overlay = await sharp({
        text: {
            text: text,
            rgba: true,
            font: 'Arial',
            dpi: 250,
        },
    }).png().toBuffer();
    await image.composite([{
        input: overlay,
        tile: true,
    }]).toFile(`./result.gif`);
};
```

**Input / Usage of Function**
```typescript
const banner = sharp(Buffer.from(arrayBuffer), {animated: true});
await overlayTextOnImage('Hello', banner);
```

### Please provide sample image(s) that help explain this question
**Original Image provided as buffer (Banner)**
![original](https://user-images.githubusercontent.com/66902977/210975410-106f5fbf-26d2-4aed-8dde-191ddf7edd7b.gif)
**Overlay Image generated in the function mentioned above**
![overlay](https://user-images.githubusercontent.com/66902977/210975498-11b703ea-3981-4961-a846-0bef98947364.png)
**Result after overlaying them**
![result](https://user-images.githubusercontent.com/66902977/210975553-2eaa4a9f-e968-45ae-b4ce-0ef9ae9a1d8a.gif)
**Result after rezizing the Text to match the size of the Image**
```typescript
const overlayTextOnImage = async (text: string, image: Sharp) => {
    const metadata = await image.metadata();
    const overlay = await sharp({
        text: {
            text: text,
            rgba: true,
            width: metadata.width,
            height: metadata.height,
            font: 'Arial',
        },
    }).png().toBuffer();
    await image.composite([{
        input: overlay,
        tile: true,
    }]).toFile(`./result2.gif`);
};
```
![result2](https://user-images.githubusercontent.com/66902977/210975715-a75e1f3f-a6b0-41d3-ae01-baf1dfb92198.gif)

**Comment by lovell** [comment]
The second example looks like either an off-by-one or rounding problem, where the `overlay` is one pixel shorter than the frame height.

When you use `tile` with animated images, you'll need to ensure all overlay image heights match the frame height to avoid the "scrolling effect" (great for 1990s dance music videos, less so for what you're trying to achieve).

Is the rendered `overlay` PNG image in the second example exactly the `metadata.width` x `metadata.height` dimensions you provided, or could it be slightly smaller? If smaller, you can use [extend](https://sharp.pixelplumbing.com/api-resize#extend) to add pixels at the top and/or bottom.

**Comment by Zueripat** [comment]
Hello @lovell 

First of all thank you for this fast reply. 

I was able to check the width and height of both background and overlay Image and was able to confirm, that their dimensions did not match. I have now corrected it using the `extent` method and achieved a non scrolling overlay.

![result](https://user-images.githubusercontent.com/66902977/211148891-107841b7-7cc1-4370-bd46-5d9aa9ba40fb.gif)

Using the [extend](https://sharp.pixelplumbing.com/api-resize#extend) method i was able to resolve both of my problems although it might be a little bit unconstitutional.

Best of wishes Patrick
